src/main/kotlin/org/rust/ide/intentions/ExtractInlineModule.kt

```diff
@@ -0,0 +1,56 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiFileFactory
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.RustLanguage
+import org.rust.lang.core.psi.RustModDeclItem
+import org.rust.lang.core.psi.RustModItem
+import org.rust.lang.core.psi.impl.RustFile
+import org.rust.lang.core.psi.util.parentOfType
+
+class ExtractInlineModule : IntentionAction {
+    override fun getFamilyName() = "Extract inline module structure"
+    override fun getText() = "Extract Inline Module"
+    override fun startInWriteAction() = true
+
+    fun createFileFromText(project: Project, text: String): RustFile? =
+        PsiFileFactory.getInstance(project).createFileFromText("DUMMY.rs", RustLanguage, text) as RustFile?
+
+    inline fun <reified T : PsiElement> createModDeclItem(project: Project, modName: String): T? {
+        val file = createFileFromText(project, "mod $modName;")
+        return PsiTreeUtil.findChildOfType(file, T::class.java)
+    }
```

Comment:
Ouch, that wasn't my intention. Have a look at [`PsiElement#parentOfType`](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/lang/core/psi/util/RustPsiExtensions.kt#L18). I meant to create a similar extension method below that one, that will find child of given type. Sth like:
```kotlin
inline fun <reified T : PsiElement> PsiElement.childOfType(): T? =
    PsiTreeUtil.findChildOfType(this, T::class.java)
```